crates/variables/src/host_component.rs

```diff
@@ -89,11 +90,41 @@ impl config::Host for ComponentConfig {
     }
 }
 
-impl From<Error> for config::Error {
+#[async_trait]
+impl spin_world::v1::config::Host for ComponentVariables {
+    async fn get_config(
+        &mut self,
+        key: String,
+    ) -> Result<Result<String, spin_world::v1::config::Error>> {
+        Ok(async {
+            // Set by DynamicHostComponent::update_data
+            let component_id = self.component_id.as_deref().unwrap();
+            let key = Key::new(&key)?;
+            Ok(self
+                .resolver
+                .get()
+                .unwrap()
+                .resolve(component_id, key)
+                .await?)
+        }
+        .await)
+    }
+}
+
+impl From<Error> for variables::Error {
+    fn from(err: Error) -> Self {
+        match err {
+            Error::InvalidName(msg) => Self::InvalidName(msg),
+            Error::Provider(msg) => Self::Provider(msg.to_string()),
+            other => Self::Other(format!("{}", other)),
+        }
+    }
+}
+
+impl From<Error> for spin_world::v1::config::Error {
     fn from(err: Error) -> Self {
         match err {
-            Error::InvalidKey(msg) => Self::InvalidKey(msg),
-            Error::InvalidSchema(msg) => Self::InvalidSchema(msg),
+            Error::InvalidName(msg) => Self::InvalidKey(msg),
             Error::Provider(msg) => Self::Provider(msg.to_string()),
             other => Self::Other(format!("{}", other)),
```

Comment:
Why do we have a catch-all pattern here instead of exhaustively matching?

Reply:
Similar to my comment below about the `Error` type, it contains variants that aren't actionable by (and will never reach) the guest code.